### PR TITLE
setup node-ipam-controller staging repo.

### PR DESCRIPTION
**Pull request by liangyuanpeng**
xref https://github.com/kubernetes/test-infra/pull/32373#discussion_r1551057954
Part of https://github.com/kubernetes-sigs/node-ipam-controller/issues/15

/cc @aojea

### CODE REVIEW

**Review by liangyuanpeng — COMMENTED** [review]
/cc @upodroid  @ameukam

File: groups/restrictions.yaml [hunk 1]
```diff
@@ -145,6 +145,7 @@ restrictions:
       - "^k8s-infra-staging-nfd@kubernetes.io$"
       - "^k8s-infra-staging-npd@kubernetes.io$"
       - "^k8s-infra-staging-sp-operator@kubernetes.io$"
+      - "^k8s-infra-staging-node-ipam-controller@kubernetes.io$"
```
Comment (aojea) [inline groups/restrictions.yaml:148]: it has to go in  sig-network/groups, the previous block

Comment (liangyuanpeng) [reply groups/restrictions.yaml]: Thanks, I put the wrong side. Updated.

**Comment by aojea** [comment]
/ok-to-test
/assign @ameukam @upodroid 

Thanks, not familiar enough to be able to approve

File: groups/sig-network/groups.yaml [hunk 2]
```diff
@@ -39,6 +39,18 @@ groups:
       - varunmar@google.com
       - zihongz@google.com
 
+  - email-id: k8s-infra-staging-node-ipam-controller@kubernetes.io
```
Comment (liangyuanpeng) [inline groups/sig-network/groups.yaml:42]: >  groups_test.go:135: Number of characters in project name "node-ipam-controller" should not exceed 18; is: 20

Seems like it's too long, max length is 18 and `node-ipam-controller` is 20, need the new email-id

Comment (ameukam) [reply groups/sig-network/groups.yaml:42]: `k8s-infra-staging-npc@kubernetes.io`? cc @aojea

File: registry.k8s.io/manifests/k8s-staging-node-ipam-controller/promoter-manifest.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# google group for gcr.io/k8s-staging-node-ipam-controller is k8s-infra-staging-node-ipam-controller@kubernetes.io
+registries:
+- name: gcr.io/k8s-staging-node-ipam-controller
+  src: true
+- name: us.gcr.io/k8s-artifacts-prod/node-ipam-controller
+  service-account: k8s-infra-gcr-promoter@k8s-artifacts-prod.iam.gserviceaccount.com
+- name: eu.gcr.io/k8s-artifacts-prod/node-ipam-controller
+  service-account: k8s-infra-gcr-promoter@k8s-artifacts-prod.iam.gserviceaccount.com
+- name: asia.gcr.io/k8s-artifacts-prod/node-ipam-controller
+  service-account: k8s-infra-gcr-promoter@k8s-artifacts-prod.iam.gserviceaccount.com
```
Comment (ameukam) [inline registry.k8s.io/manifests/k8s-staging-node-ipam-controller/promoter-manifest.yaml:10]: This is incorrect. We now use Artifact Registry repos:
https://github.com/kubernetes/k8s.io/blob/main/registry.k8s.io/manifests/k8s-staging-apisnoop/promoter-manifest.yaml

**Comment by ameukam** [comment]
Alternatively we could use an existing repo: https://github.com/kubernetes/k8s.io/tree/main/registry.k8s.io/manifests/k8s-staging-networking

**Comment by aojea** [comment]
> Alternatively we could use an existing repo: https://github.com/kubernetes/k8s.io/tree/main/registry.k8s.io/manifests/k8s-staging-networking

is there any problem on using the same repo?
it is simpler if the permissions are not too broad

File: groups/sig-network/groups.yaml [hunk 4]
```diff
@@ -39,6 +39,18 @@ groups:
...
+    name: k8s-infra-staging-node-ipam-controller
+    description: |-
+      ACL for staging node-ipam-controller images.
+    settings:
+      ReconcileMembers: "true"
+    members:
+      - ameukam@gmail.com
+      - antonio.ojea.garcia@gmail.com
+      - sarveshr@google.com
```
[the remark this answers is not in the document]
Comment (mneverov) [reply groups/sig-network/groups.yaml:51]: neverov.max@gmail.com
:open_hands:

**Comment by aojea** [comment]
then 

> Alternatively we could use an existing repo: https://github.com/kubernetes/k8s.io/tree/main/registry.k8s.io/manifests/k8s-staging-networking

@liangyuanpeng let's do this,

File: groups/sig-network/groups.yaml (hunk 2 above)
Comment (liangyuanpeng) [reply groups/sig-network/groups.yaml:42]: just for confirm that so k8s-infra-staging-node-ipam-controller@kubernetes.io is not need anymore if we use an existing repo https://github.com/kubernetes/k8s.io/tree/main/registry.k8s.io/manifests/k8s-staging-networking, right?

And how should i moving the member of 

```yaml
      - ameukam@gmail.com
      - antonio.ojea.garcia@gmail.com
      - sarveshr@google.com
      - neverov.max@gmail.com
```

@ameukam

Comment (ameukam) [reply groups/sig-network/groups.yaml:42]: Yes.

**Comment by ameukam** [comment]
@liangyuanpeng we don't need this PR anymore. we already have a staging repo to use.

**Comment by liangyuanpeng** [comment]
Got it, just using https://github.com/kubernetes/k8s.io/blob/main/registry.k8s.io/images/k8s-staging-networking/images.yaml , closing...  thanks for explain.

**Pull request closed without merging**